internal/git/git.go

```diff
@@ -144,6 +144,15 @@ func (r *Repo) GitStdin(args []string, stdin io.Reader) (string, error) {
 	return strings.TrimSpace(string(out)), nil
 }
 
+// DetachedHead returns true if the repository is in the detached HEAD.
+func (r *Repo) DetachedHead() (bool, error) {
+	name, err := r.RevParse(&RevParse{Rev: "HEAD", SymbolicFullName: true})
```

Comment:
Can we do `git symbolic-ref -q HEAD` instead? Seems more explicit. https://stackoverflow.com/questions/52221558/programmatically-check-if-head-is-detached

Reply:
Done.